### PR TITLE
adding author to aggs

### PR DESCRIPTION
**Pull request by timatron**

### CODE REVIEW

**Comment by jameswburke** [comment]
:bee:

File: lib/class-sp-wp-search.php [hunk 1]
```diff
@@ -305,6 +305,17 @@ public static function wp_to_es_args( $args ) {
 						);
 
 						break;
+
+					case 'author':
+						$es_query_args['aggregations'][ $label ] = array(
+							'terms' => array(
+								'field' => $facet['field'],
```
Comment (mboynes) [inline lib/class-sp-wp-search.php]: Why did you choose to make this dynamic?

**Comment by mboynes** [comment]
This is only half the work, you would also need to add a parser on the other end in `get_facet_data()`

File: lib/class-sp-wp-search.php (hunk 1 above)
Comment (mboynes) [inline lib/class-sp-wp-search.php]: From my perspective, it should be static. You need to either choose the login, nicename, or the user id. For the taxonomies we are using `slug`, so `post_author.login` would probably be the most consistent with that.

Comment (timatron) [inline lib/class-sp-wp-search.php]: I made it dynamic to account for plugins like CAP and Bylines. No?

**Comment by mboynes** [comment]
I'm in favor of adding author as a default facet. To complete this:

* [ ] change the field
* [ ] finish the other half of the implementation
* [ ] add unit tests

File: lib/class-sp-wp-search.php (hunk 1 above)
Comment (mboynes) [inline lib/class-sp-wp-search.php]: nah, CAP and bylines are both tax terms, so there's already a built-in facet for those

**Comment by mboynes** [comment]
You should update this to use the user's login [as mentioned here](https://github.com/alleyinteractive/searchpress/pull/33#discussion_r78389722) instead of user id to be consistent with how taxonomy terms work. The unit test should also test that it's really working (test the data), not just the existence of the bucket (see all the other tests in that file)

**Pull request closed without merging**